server/backends/github/github.go

```diff
@@ -66,6 +66,12 @@ func AuthEnabled(env environment.Env) bool {
 	return *JwtKey != ""
 }
 
+func AlwaysEnableStatusReporting() bool {
+	// If an on-prem customer has hard-coded an access token for GitHub status
```

Comment:
(OSS users can use it too)

```suggestion
	// If the user has hard-coded an access token for GitHub status
```